src/main/java/ch/wisv/areafiftylan/seats/service/SeatServiceImpl.java

```diff
@@ -52,80 +52,73 @@ public SeatServiceImpl(SeatRepository seatRepository, TeamService teamService, T
-            return false;
-        } else {
-            reserveSeat(seat, ticketId);
-            return true;
-        }
+    public Seat getSeatBySeatGroupAndSeatNumber(String groupName, int seatNumber) {
+        return seatRepository.findBySeatGroupAndSeatNumber(groupName, seatNumber)
+                .orElseThrow(SeatNotFoundException::new);
     }
 
     @Override
-    public void reserveSeatForAdmin(String seatGroup, int seatNumber, Long ticketId) {
-        Seat seat = getSeatBySeatGroupAndSeatNumber(seatGroup, seatNumber);
-        reserveSeat(seat, ticketId);
-    }
-
-    /**
-     * Reserves a Seat, and sets the Ticket to ticketId's Ticket.
-     * Also frees any previous Seat belonging to the Ticket.
-     * This method ignores any current user assigned to the seat, that logic must be done before calling this class.
-     * If any user was assigned this seat before, send an email to let them know their seat was overridden.
-     * This method is synchronized to prevent the extremely unlikely event of simultaneous requests.
-     *
-     * @param seat The Seat to reserve.
-     * @param ticketId The Id of the Ticket to be set to the Seat.
-     */
     @Synchronized
-    private void reserveSeat(Seat seat, Long ticketId) {
+    public boolean reserveSeat(String groupName, int seatNumber, Long ticketId, boolean allowSeatOverride) {
+        Seat seat = getSeatBySeatGroupAndSeatNumber(groupName, seatNumber);
+        Ticket ticket = null;
+
+        if (!allowSeatOverride && (seat.isTaken() || seat.isLocked() || ticketId == null)) {
+            return false;
+        }
+
+        if (seat.isTaken() && seat.getTicket().getOwner() != null) {
+            mailService.sendSeatOverrideMail(seat.getTicket().getOwner());
+        }
         if (ticketId != null) {
             seatRepository.findByTicketId(ticketId).ifPresent(previousSeat -> previousSeat.setTicket(null));
-            Ticket ticket = ticketService.getTicketById(ticketId);
+            ticket = ticketService.getTicketById(ticketId);
             if (!ticket.isValid()) {
                 throw new InvalidTicketException("Unable to reserve seat for an invalid Ticket");
             }
-            if (seat.getTicket() != null && seat.getTicket().getOwner() != null) {
-                mailService.sendSeatOverrideMail(seat.getTicket().getOwner());
-            }
-            seat.setTicket(ticket);
-        } else {
-            seat.setTicket(null);
         }
-        seat.setTaken(true);
+        seat.setTicket(ticket);
         seatRepository.saveAndFlush(seat);
+        return true;
     }
 
     @Override
-    public Seat getSeatBySeatGroupAndSeatNumber(String groupName, int seatNumber) {
-        return seatRepository.findBySeatGroupAndSeatNumber(groupName, seatNumber)
-                .orElseThrow(SeatNotFoundException::new);
+    public void clearSeat(String groupName, int seatNumber) {
+        reserveSeat(groupName, seatNumber, null, true);
```

Comment:
Correct. There is now only one method that modifies seats (reserveSeat()), and that has an email check built in.